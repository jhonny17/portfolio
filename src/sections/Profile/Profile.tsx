import cx from 'classnames';
import Image from 'next/image';

import profilePicture from '@/assets/images/profile_photo_gray.webp';

import { Card } from '@/components/ui/Card';
import { GlowBackdrop } from '@/components/ui/GlowBackdrop';

export const Profile = () => {
  return (
    <section
      className={cx(
        'grid',
        'sm:grid-cols-[1fr,auto]',
        'lg:grid-cols-[1fr,1fr]',
        'items-center',
        'w-full',
        'relative',
        'overflow-hidden',
      )}
    >
      <GlowBackdrop
        className={cx(
          'h-full',
          'sm:w-7/12',
          'mt-auto',
          'mx-auto',
          'sm:ml-0',
          'sm:mr-auto',
          'sm:my-auto',
          'sm:rounded-r-full',
        )}
      />

      <div className={cx('p-8', 'order-2', 'sm:order-1')}>
        <Card className={cx('grid', 'mx-auto', 'sm:w-[390px]', 'md:w-[450px]')}>
          <div className={cx('mb-2', 'text-center', '*:text-gray-400')}>
            <h2 className={cx('font-logo', 'text-2xl', 'mb-1')}>
              Jhonny Vargas Arias
            </h2>
            <div className="text-sm">Software Engineer</div>
          </div>
        </Card>
      </div>

      <div
        className={cx(
          'order-1',
          'sm:order-2',
          'bg-neutral-950',
          'sm:bg-transparent',
        )}
      >
        <Image
          src={profilePicture}
          alt="Jhonny Vargas Arias"
          width={300}
          height={300}
          decoding="async"
          className={cx(
            'aspect-square',
            'ml-auto',
            'mt-8',
            'w-[180px]',
            'md:w-[300px]',
          )}
        />
      </div>
    </section>
  );
};
