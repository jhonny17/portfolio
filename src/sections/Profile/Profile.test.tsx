import { expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';

import { Profile } from './Profile';

it('renders the profile section', () => {
  render(<Profile />);

  const profileSection = screen.getByRole('heading', {
    level: 2,
    name: /Jhonny Vargas Arias/i,
  });

  expect(profileSection).toBeInTheDocument();

  const profileRole = screen.getByText(/Software Engineer/i);
  expect(profileRole).toBeInTheDocument();

  const profilePicture = screen.getByAltText(/Jhonny Vargas Arias/i);
  expect(profilePicture).toBeInTheDocument();
});

it('matches the snapshot', () => {
  const { asFragment } = render(<Profile />);
  expect(asFragment()).toMatchSnapshot();
});
