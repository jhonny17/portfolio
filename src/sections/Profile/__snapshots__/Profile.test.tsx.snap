// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="grid sm:grid-cols-[1fr,auto] lg:grid-cols-[1fr,1fr] items-center w-full relative overflow-hidden"
  >
    <div
      class="z-[-1] absolute inset-0 blur-3xl bg-gradient-to-r from-emerald-900 from-0% to-emerald-800 to-100% h-full sm:w-7/12 mt-auto mx-auto sm:ml-0 sm:mr-auto sm:my-auto sm:rounded-r-full"
    />
    <div
      class="p-8 order-2 sm:order-1"
    >
      <div
        class="p-4 md:p-8 shadow-lg rounded-lg backdrop-sepia-0 backdrop-blur-sm bg-slate-950/40 grid mx-auto sm:w-[390px] md:w-[450px]"
      >
        <div
          class="mb-2 text-center *:text-gray-400"
        >
          <h2
            class="font-logo text-2xl mb-1"
          >
            Jhonny Vargas Arias
          </h2>
          <div
            class="text-sm"
          >
            Software Engineer
          </div>
        </div>
      </div>
    </div>
    <div
      class="order-1 sm:order-2 bg-neutral-950 sm:bg-transparent"
    >
      <img
        alt="Jhonny Vargas Arias"
        class="aspect-square ml-auto mt-8 w-[180px] md:w-[300px]"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fsrc%2Fassets%2Fimages%2Fprofile_photo_gray.webp&w=640&q=75"
        srcset="/_next/image?url=%2Fsrc%2Fassets%2Fimages%2Fprofile_photo_gray.webp&w=384&q=75 1x, /_next/image?url=%2Fsrc%2Fassets%2Fimages%2Fprofile_photo_gray.webp&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
  </section>
</DocumentFragment>
`;
